db/version_set.cc

```diff
@@ -2380,6 +2429,181 @@ void Version::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
   }
 }
 
+#ifdef USE_COROUTINES
+Status Version::ProcessBatch(
+    const ReadOptions& read_options, FilePickerMultiGet* batch,
+    std::vector<folly::coro::Task<Status>>& mget_tasks,
+    std::unordered_map<uint64_t, BlobReadContexts>* blob_ctxs,
+    autovector<FilePickerMultiGet, 4>& batches, std::queue<size_t>& waiting,
```

Comment:
What's the `waiting` queue used for?

Reply:
Its used for an optimization on line 2489 - if there are no waiting batches, that means this is the only batch and we can possibly skip the coroutine and call the synchronous function.